src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -300,8 +300,8 @@ class FillExtrusionBucket implements Bucket {
     }
 
     addFeature(feature: BucketFeature, geometry: Array<Array<Point>>, index: number, canonical: CanonicalTileID, imagePositions: {[_: string]: ImagePosition}) {
-        const flatRoof = this.enableTerrain && feature.properties && feature.properties.hasOwnProperty('type') &&
-            feature.properties.hasOwnProperty('height') && vectorTileFeatureTypes[feature.type] === 'Polygon';
+        const flatRoof = this.enableTerrain && feature.properties &&
```

Comment:
I think you can get rid of `feature.properties && vectorTileFeatureTypes[feature.type] === 'Polygon'` as well,
requiring properties for flat tops is weird.
Currently we require feature-type `Polygon` as a requirement tessellating and filling out the roof. But if someone is using LineString to have `open-top` extrusions, I don't see why we should exclude them from being flat tops.

Reply:
I think that ties to @astojilj's comment here https://github.com/mapbox/mapbox-gl-js/pull/10347#discussion_r568484982.

>Current code is a conservative hack (best effort) to enable flat roofs for some of the buildings.
Don't think it is a good idea to enable it for any feature that is larger than the building (countries, routes)

I'd be concerned about the behavior if we begin to be even less conservative here. The main issue is what has exposed #10170, we should fix that case first, and address your suggestion if needed. I'll hold on this suggestion.